slangpy/core/calldata.py

```
@@ -519,28 +519,38 @@ def _try_build_shader(
+                        and hit_group.intersection_entry_point not in entry_point_names
+                    ):
+                        entry_point_names.add(hit_group.intersection_entry_point)
                 for miss_entry_point in build_info.ray_tracing_miss_entry_points:
-                    eps.append(build_info.module.device_module.entry_point(miss_entry_point))
+                    if miss_entry_point not in entry_point_names:
+                        entry_point_names.add(miss_entry_point)
+                for callable_entry_point in build_info.ray_tracing_callable_entry_points:
+                    if callable_entry_point not in entry_point_names:
+                        entry_point_names.add(callable_entry_point)
+                for entry_point in entry_point_names:
+                    eps.append(build_info.module.device_module.entry_point(entry_point))
```

Ruff (0.15.12)

[warning] Use `list.extend` to create a transformed list

(PERF401)